### PR TITLE
Update doc of elementwise functions

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR suggests to update the documentation of `parameters` of element-wise functions by adding `broadcasting` information to align with Numpy docs.

- [X] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_trigonometric.py [hunk 1]
```diff
@@ -2287,7 +2315,8 @@ def reduce_hypot(x, /, *, axis=None, dtype=None, keepdims=False, out=None):
     Input array, expected to have numeric data type.
 out : {None, dpnp.ndarray, usm_ndarray}, optional
     Output array to populate.
-    Array must have the correct shape and the expected data type.
+    Array must have a shape that the input broadcast to and
```
Comment (vtavana) [inline dpnp/dpnp_iface_trigonometric.py:2318]:
```suggestion
    The array must have a shape to which the input can be broadcast and
```

However, it seems that `dpnp` (and `dpctl`) do not support broadcasting for unary functions.

```py
import numpy, dpnp
a = numpy.ones((2, 3))
out1 = numpy.empty((4, 2, 3))
numpy.tanh(a, out=out1).shape
# (4, 2, 3)

b = dpnp.ones((2, 3))
out2 = dpnp.empty((4, 2, 3))
dpnp.tanh(b, out=out2).shape
# ValueError: The shape of input and output arrays are inconsistent. Expected output shape is (2, 3), got (4, 2, 3)
```

File: dpnp/dpnp_iface_mathematical.py [hunk 2]
```diff
@@ -2486,7 +2507,8 @@ def gradient(f, *varargs, axis=None, edge_order=1):
     Both inputs `x1` and `x2` can not be scalars at the same time.
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply dpnp/dpnp_iface_mathematical.py:2502]: I have added information about this, thank you !

File: dpnp/dpnp_iface_trigonometric.py (hunk 1 above)
Comment (vlad-perevezentsev) [reply dpnp/dpnp_iface_trigonometric.py]: Nice point!
I have removed the changes for `out` parameter

File: dpnp/dpnp_iface_mathematical.py [hunk 3]
```diff
@@ -2576,7 +2594,8 @@ def gradient(f, *varargs, axis=None, edge_order=1):
     Input array, expected to have numeric data type.
 out : {None, dpnp.ndarray, usm_ndarray}, optional
     Output array to populate.
-    Array must have the correct shape and the expected data type.
+    Array must have a shape that the input broadcast to and
+    the expected data type.
```
Comment (vtavana) [inline dpnp/dpnp_iface_mathematical.py:2598]: This one was missed.

```suggestion
    Array must have the correct shape and the expected data type.
```

**Review by vtavana — APPROVED** [review]
Thank you, @vlad-perevezentsev!

**Pull request merged**